packages/core/src/node/process/terminal-process.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { injectable, inject } from 'inversify';
+import * as stream from 'stream';
+import { ILogger } from '../../common/logger';
+import { Process } from './process';
+
+const pty = require("node-pty");
+
+export const TerminalProcessOptions = Symbol("TerminalProcessOptions");
+export interface TerminalProcessOptions {
+    command: string,
+    args?: string[],
+    options?: object
+}
+
+export const TerminalProcessFactory = Symbol("TerminalProcessFactory");
+export type TerminalProcessFactory = (options: TerminalProcessOptions) => TerminalProcess;
+
+/* Use this instead of the node-pty stream, since the node-pty stream is already resumed.  */
+export class TerminalReadableStream extends stream.Readable {
+    constructor(protected readonly terminal: any, opts?: any) {
+        super(opts);
+        this.terminal.on('data', (data: any) => {
+            this.push(data);
+        });
+    }
+    _read(size: number) {
```

Comment:
this is from:
```
Custom Readable streams must call the new stream.Readable([options]) constructor and implement the readable._read() method.
```
See https://nodejs.org/dist/latest-v8.x/docs/api/stream.html#stream_implementing_a_readable_stream